### PR TITLE
Added savepoint and friends

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -4347,96 +4379,84 @@ set_list:
...
-    $$ = &SetVarExpr{Name: NewColName(string($1)), Expr: $2, Scope: SetScope_Session}
+    $$ = SetScope_Persist
+  }
+| PERSIST_ONLY
+  {
+    $$ = SetScope_PersistOnly
+  }
+
+set_expression_assignment:
+  column_name '=' ON
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y:4449]: Do we take true and false as well?

Comment (zachmu) [reply go/vt/sqlparser/sql.y:4449]:
```
mysql> set autocommit = false;
Query OK, 0 rows affected (0.02 sec)
```

File: go/vt/sqlparser/token.go [hunk 2]
```diff
@@ -358,7 +358,7 @@ var keywords = map[string]int{
 	"real":                REAL,
 	"references":          REFERENCES,
 	"regexp":              REGEXP,
-	"release":             UNUSED,
+	"release":             RELEASE,
```
Comment (zachmu) [inline go/vt/sqlparser/token.go:361]: Don't forget to add these new guys to one of the reserved / non reserved keyword lists

**Review by zachmu — APPROVED** [review]
LG thanks for doing this

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (Hydrocharged) [reply go/vt/sqlparser/sql.y:4449]: Yep, that's handled by `expression`

**Pull request merged**